salt/utils/minions.py

```diff
@@ -1156,11 +1156,28 @@ def mine_get(tgt, fun, tgt_type='glob', opts=None):
             tgt_type)
     minions = _res['minions']
     cache = salt.cache.factory(opts)
+
+    _ret_dict = False
+    if isinstance(fun, six.string_types):
+        functions = list(set(fun.split(',')))
+        if len(functions) > 1:
+            _ret_dict = True
+    elif isinstance(fun, list):
+        functions = fun
+        _ret_dict = True
+    else:
+        return {}
+
     for minion in minions:
         mdata = cache.fetch('minions/{0}'.format(minion), 'mine')
         if mdata is None:
             continue
-        fdata = mdata.get(fun)
-        if fdata:
-            ret[minion] = fdata
+
+        if isinstance(mdata, dict):
+            for fun in functions:
+                if mdata.has_key(fun):
+                    if _ret_dict:
+                        ret.setdefault(fun, {})[minion] = mdata.get(fun)
+                    else:
+                        ret[minion] = mdata.get(fun)
```

Comment:
If this works for multi-function jobs, isn't this overwriting itself, being valid only for the last function?

If it's just for single-function jobs, why is inside the for?

Reply:
Moved single-function jib out of the loop. Is it better now?